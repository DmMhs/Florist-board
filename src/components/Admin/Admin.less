.Admin {
  min-height: calc(100vh - 10rem);
  h1 {
    font-weight: normal;
    color: #333333;
  }
  hr {
    border: 0.5px solid #eee;
    width: 50%;
  }
  .admin-actions {
    list-style: none;
    display: flex;
    justify-content: center;
    align-items: center;
    li {
      font-weight: normal;
      color: #333333;
      font-size: 1.3rem;
      margin: 8px;
      background-color: whitesmoke;
      padding: 8px;
      cursor: pointer;
      &:hover {
        background-color: #ccc;
      }
    }
  }
}
