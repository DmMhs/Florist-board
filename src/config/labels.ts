const labels = {
  brand: 'FLORIST.UA',
  adress: 'MARKET SQUARE, LVIV, UA',
  postCode: '346346',
  phone: '+123-45-678',
  navigation: {
    home: 'HOME',
    shop: 'SHOP',
    gallery: 'GALLERY',
    contacts: 'CONTACTS',
    cart: 'CART',
    account: {
      main: 'MY ACCOUNT',
      menu: {
        signIn: 'SIGN IN',
        signUp: 'SIGN UP'
      }
    }
  }
};

export default labels;
