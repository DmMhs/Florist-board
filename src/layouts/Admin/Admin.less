.Admin {
  min-height: calc(100vh - 10rem);
  color: #333333;
  h1 {
    font-weight: normal;
  }
  hr {
    border: 0.5px solid #eee;
    width: 50%;
  }
  .accent {
    font-weight: 600;
  }
  .admin-actions {
    list-style: none;
    display: flex;
    justify-content: center;
    align-items: center;
    li {
      font-weight: normal;
      color: #333333;
      font-size: 1.3rem;
      margin: 8px;
      background-color: whitesmoke;
      padding: 8px;
      cursor: pointer;
      &:hover,
      &.active {
        background-color: #ccc;
      }
      a {
        display: inline-block;
        width: 100%;
        height: 100%;
      }
    }
  }
  .form-wrapper {
    margin: 2rem 0;
    .form {
      width: 50%;
      border: 1px solid #eee;
      border-radius: 7px;
      padding: 1.5rem;
      display: flex;
      flex-direction: column;
      align-items: center;
      margin: 2rem auto;
      hr {
        border: 0.5px solid #eee;
        width: 50%;
      }
      .form-control {
        input {
          width: 40%;
        }
        textarea {
          min-height: 4rem;
          width: 30%;
        }
        .price {
          display: flex;
          width: 40%;
          input {
            width: 100%;
          }
        }
      }
      button {
        margin-top: 1rem;
        background-color: #eee;
        color: #333333;
        border: none;
        cursor: pointer;
        padding: 8px;
        font-size: 1.2rem;
        &:hover {
          background-color: #ccc;
        }
      }
    }
  }
}

@media screen and (max-width: 768px) {
  .Admin {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    box-sizing: border-box;
    h1 {
      font-size: 1.7rem;
      margin-top: 3rem;
    }
    .admin-actions {
      width: 100%;
      flex-direction: column;
      padding: 0;
      box-sizing: border-box;
      li {
        font-size: 1.2rem;
        width: 50%;
      }
    }
    .form-wrapper {
      width: 90vw;
      box-sizing: border-box;
      display: flex;
      flex-direction: column;
      align-items: center;
      .form {
        width: 70vw;
        margin: 0 auto;
        .form-control {
          box-sizing: border-box;
          input,
          textarea {
            width: 100%;
          }
          p {
            font-size: 0.9rem;
            &.current {
              font-size: 0.8rem;
            }
          }
          textarea {
            min-height: 4rem;
          }
          .price {
            width: 100%;
            box-sizing: border-box;
            input {
              width: 100%;
              padding: 0;
              line-height: 1.7rem;
            }
          }
        }
      }
    }
  }
}
