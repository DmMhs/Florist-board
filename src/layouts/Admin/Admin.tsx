import React, { Component } from 'react';
import firebase from 'firebase';

import './Admin.less';
import { AppContext } from '../../AppContext';
import { RouteComponentProps, withRouter, Redirect } from 'react-router';
import AddProduct from './AddProduct/AddProduct';
import AddGalleryImage from './AddGalleryImage/AddGalleryImage';
import ChangeLabels from './ChangeLabels/ChangeLabels';
import ChangeContacts from './ChangeContacts/ChangeContacts';
import ChangeURLs from './ChangeURLs/ChangeURLs';

interface AdminState {
  mode:
    | 'add-product'
    | 'edit-product'
    | 'configurate-gallery'
    | 'configurate-labels' 
    | 'configurate-urls' 
    | 'configurate-contacts';
}

class Admin extends Component<RouteComponentProps<{}>, AdminState> {
  constructor(props: RouteComponentProps<{}>) {
    super(props);
    this.state = {
      mode: 'add-product'
    };
  }

  private switchModeTo = (
    mode:
      | 'add-product'
      | 'edit-product'
      | 'configurate-gallery'
      | 'configurate-labels'
      | 'configurate-urls'
      | 'configurate-contacts'
  ) => {
    this.setState({
      mode
    });
  };

  public render() {
    const { mode } = this.state;
    const context = this.context;
    const lang = context.state.lang;
    const labels = context.state.labels;
    const labelsRoot = labels[lang].pages.admin;

    let form: JSX.Element;
    switch (mode) {
      case 'add-product':
        form = <AddProduct />;
        break;
      case 'edit-product':
        form = <h3>Edit Product Form</h3>;
        break;
      case 'configurate-gallery':
        form = <AddGalleryImage />;
        break;
      case 'configurate-labels':
        form = <ChangeLabels />;
        break;
      case 'configurate-urls':
        form = <ChangeURLs />;
        break;
      case 'configurate-contacts':
        form = <ChangeContacts />;
        break;  
      default:
        form = <h3>Hm...</h3>;
    }
    return (
      <AppContext.Consumer>
        {value =>
          value && value.state.userRole === 'admin' ? (
            <div className="Admin">
              <h1>{labelsRoot.title}</h1>
              <hr />
              <ul className="admin-actions">
                <li>
                  <a onClick={this.switchModeTo.bind(this, 'add-product')}>
                    {labelsRoot.navigation.addProduct}
                  </a>
                </li>
                <li>
                  <a
                    onClick={this.switchModeTo.bind(
                      this,
                      'configurate-gallery'
                    )}
                  >
                    {labelsRoot.navigation.galleryImages}
                  </a>
                </li>
                <li>
                  <a
                    onClick={this.switchModeTo.bind(this, 'configurate-labels')}
                  >
                    {labelsRoot.navigation.labels}
                  </a>
                </li>
                <li>
                  <a
                    onClick={this.switchModeTo.bind(this, 'configurate-urls')}
                  >
                    {labelsRoot.navigation.urls}
                  </a>
                </li>
                <li>
                  <a
                    onClick={this.switchModeTo.bind(this, 'configurate-contacts')}
                  >
                    {labelsRoot.navigation.contacts}
                  </a>
                </li>
              </ul>
              <div className="form-wrapper">{form}</div>
            </div>
          ) : (
            <Redirect to="/" />
          )
        }
      </AppContext.Consumer>
    );
  }
}

Admin.contextType = AppContext;

export default withRouter<RouteComponentProps<{}>>(Admin);
