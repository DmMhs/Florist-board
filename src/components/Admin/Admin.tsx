import React, { Component } from 'react';
import firebase from 'firebase';

import './Admin.less';
import { AppContext } from '../../AppContext';
import { RouteComponentProps, withRouter, Redirect } from 'react-router';
import AddProduct from './AddProduct/AddProduct';
import labels from '../../config/labels';
import AddGalleryImage from './AddGalleryImage/AddGalleryImage';

interface AdminState {
  mode:
    | 'add-product'
    | 'edit-product'
    | 'configurate-gallery'
    | 'configurate-labels';
}

class Admin extends Component<RouteComponentProps<{}>, AdminState> {
  constructor(props: RouteComponentProps<{}>) {
    super(props);
    this.state = {
      mode: 'add-product'
    };
  }

  private switchModeTo = (
    mode:
      | 'add-product'
      | 'edit-product'
      | 'configurate-gallery'
      | 'configurate-labels'
  ) => {
    this.setState({
      mode
    });
  };

  public render() {
    const { mode } = this.state;

    let form: JSX.Element;
    switch (mode) {
      case 'add-product':
        form = <AddProduct />;
        break;
      case 'edit-product':
        form = <h3>Edit Product Form</h3>;
        break;
      case 'configurate-gallery':
        form = <AddGalleryImage />;
        break;
      case 'configurate-labels':
        form = <h3>Labels Form</h3>;
        break;
      default:
        form = <h3>Default form</h3>;
    }
    return (
      <AppContext.Consumer>
        {value =>
          value && value.state.userRole === 'admin' ? (
            <div className="Admin">
              <h1>{labels[value.state.lang as string].pages.admin.title}</h1>
              <hr />
              <ul className="admin-actions">
                <li>
                  <a onClick={this.switchModeTo.bind(this, 'add-product')}>
                    Add Product
                  </a>
                </li>
                <li>
                  <a
                    onClick={this.switchModeTo.bind(
                      this,
                      'configurate-gallery'
                    )}
                  >
                    Gallery Images
                  </a>
                </li>
                <li>
                  <a
                    onClick={this.switchModeTo.bind(this, 'configurate-labels')}
                  >
                    Labels
                  </a>
                </li>
              </ul>
              <div className="form-wrapper">{form}</div>
            </div>
          ) : (
            <Redirect to="/" />
          )
        }
      </AppContext.Consumer>
    );
  }
}
export default withRouter<RouteComponentProps<{}>>(Admin);
