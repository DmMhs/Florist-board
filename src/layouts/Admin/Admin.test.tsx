import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';

import Admin from './Admin';
import AppContextProvider from '../../AppContext';
import labels from '../../config/labels';
import { createMemoryHistory } from 'history';

describe('Admin works as expected', () => {
  it('Admin component matches a snapshot', () => {
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Admin />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      userLogin: 'testLogin',
      userId: 'testId',
      userToken: 'testToken',
      userRole: 'admin',
      userAuthenticated: false,
      authenticationMethod: undefined,
      lang: 'en',
      labels: labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
  it('mode switching changes the state', () => {
    const match = { params: { mode: 'add-product' } };
    const historyMock = { replace: jest.fn() };
    const wrapper = mount(
      <BrowserRouter>
        <AppContextProvider>
          <Admin.WrappedComponent match={match} history={historyMock} />
        </AppContextProvider>
      </BrowserRouter>
    );
    const context = wrapper.find('AppContextProvider').instance();
    context.setState({
      userLogin: 'testLogin',
      userId: 'testId',
      userToken: 'testToken',
      userRole: 'admin',
      userAuthenticated: true,
      authenticationMethod: undefined,
      lang: 'en',
      labels,
      fetchInProgress: false,
      mobileMode: true,
      showNavigation: false,
      togglePosition: 'absolute'
    });
    wrapper.update();
    wrapper.find('.configurateGallery').simulate('click');
    expect(historyMock.replace.mock.calls[0]).toEqual(['configurate-gallery']);
  });
});
