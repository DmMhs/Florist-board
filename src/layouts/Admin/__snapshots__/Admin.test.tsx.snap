// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin works as expected Admin component matches a snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(AppContextProvider)>
      <Route>
        <AppContextProvider
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <withRouter(Admin)>
            <Route>
              <Admin
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div
                  className="Admin"
                >
                  <h1>
                    Admin Control Panel
                  </h1>
                  <hr />
                  <ul
                    className="admin-actions"
                  >
                    <li
                      className=""
                    >
                      <a
                        className="addProduct"
                        onClick={[Function]}
                      >
                        ADD PRODUCT
                      </a>
                    </li>
                    <li
                      className=""
                    >
                      <a
                        className="configurateGallery"
                        onClick={[Function]}
                      >
                        GALLERY IMAGES
                      </a>
                    </li>
                    <li
                      className=""
                    >
                      <a
                        className="configurateLabels"
                        onClick={[Function]}
                      >
                        LABELS
                      </a>
                    </li>
                    <li
                      className=""
                    >
                      <a
                        className="configurateURLs"
                        onClick={[Function]}
                      >
                        URLs
                      </a>
                    </li>
                    <li
                      className=""
                    >
                      <a
                        className="configurateContacts"
                        onClick={[Function]}
                      >
                        CONTACTS
                      </a>
                    </li>
                  </ul>
                  <div
                    className="form-wrapper"
                  />
                </div>
              </Admin>
            </Route>
          </withRouter(Admin)>
        </AppContextProvider>
      </Route>
    </withRouter(AppContextProvider)>
  </Router>
</BrowserRouter>
`;
